### PR TITLE
feat: add feedback component

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR adds a user feedback component to the website. It's added via a global hovering "feedback" button and a form that opens at the bottom of the page.

## How it works

This change adds a small 'feedback'-button that's stuck to the right-hand side of the screen. When interacted with, it'll open a short 3-step questionnaire for the user to fill out. When it's completed, this feedback posts the data to a url where we'll process it.

### Docusaurus

The more work that we do with custom doc components, the more valuable this'll become. For something as complicated as this, it was pretty necessary.

### TypeScript

The new component is built with TS instead of JS. To accomodate that, I had to add some extra packages to docusaurus. Old components have not been converted.

### Theme changes

I've change the primary theme color for the light theme to Unleash purple (TM) and adjusted the primary color gradient to fit this. Visually, this means:
- the search interface is now white and purple instead of white and green
- the sidebar now highlights the current page / config in purple instead of green

For the latter bit, the green was almost indistinguishable from the regular text color, so I consider this a significant improvement.

### Workflow updates

I've also added some environment variables to the github workflow to facilitate using Unleash in the docs. It's ... not working as I intended just yet (see below).

## Important files / how the feature is structured

The main changes live in the `website/src/components/UserFeedback` directory. In particular `index.tsx` is the main file. I apologize: it's kind of large and unwieldy.

`website/src/theme/Root.tsx` is the other important file. This component wraps the entire site and is where I instantiate and use Unleash and include the `UserFeedback` component.

## Unimportant files

Because I've added storybook, there's also a bunch of new files in the `stories` subdirectory. These are all standard files and can be ignored (they do not need to be vetted).

## What I'd like some feedback on

I've Used the JS proxy client to hook the new feature up to Unleash. Because docusaurus is statically generated (and then server-rendered?) I ran into some issues with Unleash not getting access to fetch. I spent some time trying to make it use node-fetch if `window.fetch` isn't available, but gave up in the end. I don't think it matters much overall, but ...

I'd love to know whether this seems like a sensible way to set it up and to use it. And also whether you've got any suggestions to make it better?

I originally looked at the React client, but found the regular proxy client easier to use for this purpose.

## Problems

At the moment, it doesn't work correctly on the vercel preview (because Unleash isn't getting a valid URL to start up). It seems that the env variables either aren't being set correctly OR that they aren't picked up by the build. I'll look more into this as soon as I can, but any tips are also much appreciated.

### CODE REVIEW

File: website/src/theme/Root.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import UF from '@site/src/components/UserFeedback';
+
+// Default implementation, that you can customize
+function Root({ children }) {
+    return (
+        <>
```
[the remark this answers is not in the document]
Comment (thomasheartman) [reply website/src/theme/Root.js:32]: @sonatype-lift ignore

File: website/src/theme/Root.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+            <UF />
+            {children}
+        </>
```
[the remark this answers is not in the document]
Comment (thomasheartman) [reply website/src/theme/Root.js:35]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (thomasheartman) [reply website/src/theme/Root.js:35]: @sonatype-lift ignore

**Comment by thomasheartman** [comment]
[Create documentation feedback component](https://trello.com/c/bksfZRah/808-create-documentation-feedback-component)

**Comment by thomasheartman** [comment]
[Create documentation feedback component](https://trello.com/c/bksfZRah/808-create-documentation-feedback-component)

**Review by chriswk — APPROVED** [review]
I love these learning experiences, keeping the entire form state in localStorage wouldn't have occurred to me. I would've thought I needed a backend to store it.

I know you said it become a bit bigger than you'd like but at least the code for the Feedback form is clear and easy to follow.


A huge thumbsup from me :+1:

**Pull request merged**